Remove unused GRPC packages

mikkamp opened the pull request:
### Changes proposed in this Pull Request:

This PR removes the unused GRPC composer packages. They are added as dependency of `googleads/google-ads-php`, however since we opted to use REST request we do not use this package. Removing the packages reduces the final plugin zip file.

Closes #1643 

### Detailed test instructions:
1. Remove and previous copy of the vendors folder and run `composer install`.
2. Confirm that the packages `grpc/grpc` and `google/grpc-gcp` are no longer found in the vendor folder.
3. Send some test requests to the Google Ads API such as load the GLA dashboard screen (make sure we have a MC and Ads account onboarded).
4. Run the unit tests locally and confirm we don't have any warnings/errors.
5. Check all the tests of this PR passed and the bundle size decreased.
![image](https://user-images.githubusercontent.com/11388669/201679608-7037b704-df56-4e61-9a02-3a40ded7562f.png)


### Changelog entry
* Tweak - Remove unused GRPC packages.

mikkamp committed:
Remove grpc packages

puntope commented on composer.json:
```diff
@@ -29,6 +29,10 @@
 		"wp-coding-standards/wpcs": "^2.3",
 		"yoast/phpunit-polyfills": "^1.0"
 	},
+	"replace" : {
+		"google/grpc-gcp": "*",
+		"grpc/grpc": "*"
+	},
```
❓  Wondering why are we using this syntax instead of just adding the `rm -rf` command into [cleanup-vendor-files.sh](https://github.com/woocommerce/google-listings-and-ads/blob/develop/bin/cleanup-vendor-files.sh)

IMO I think it's better to maintain this cleanups in the sh file to have a single source of truth. Plus the fact that replace `*`doesn't seems very declarative for me. I needed to research in order to understand that '*' will make the package version to match "everything" and hence composer won't install the package.

In any case... if we are using this strategy... then we can get rid of the Line 4 or an output will happen on composer install

```
rm: vendor/google/grpc-gcp/cloudprober/bins/opt/grpc_php_plugin: No such file or directory
```

mikkamp replied:
Great catch on the extra line to remove.

I personally think preventing a package from being installed is better than installing and removing. This method of replacing the package is something [we were doing before](https://github.com/woocommerce/google-listings-and-ads/blob/0d68e01fdf8acfa8d3faeccc627b46e82d867f1e/composer.json#L30), so it's not a new concept. We can add a note in the composer.json file if it's helpful to understand what it does.

puntope reviewed (changes requested):
One of this 2 actions should be done before approval:

a. Remove L4 from [cleanup-vendor-files.sh](https://github.com/woocommerce/google-listings-and-ads/blob/develop/bin/cleanup-vendor-files.sh)

b. Remove the vendor files inside [leanup-vendor-files.sh](https://github.com/woocommerce/google-listings-and-ads/blob/develop/bin/cleanup-vendor-files.sh) instead of using `replace: *` strategy.

puntope commented:
Thanks @mikkamp for working on this.

I left a comment in regards one issue I think needs to be addressed.

mikkamp committed:
Remove cleanup for not installed grpc-gcp package

mikkamp commented:
Thanks for the review @puntope 
I added [a comment](https://github.com/woocommerce/google-listings-and-ads/pull/1767#discussion_r1025000260) about why I think it's best to prevent the packages from being installed. I tried adding a comment in `composer.json` but I got the following error:

> Comments are not permitted in JSON.

So I decided to leave it out, it should be pretty easy to find the attached commit/PR where the explanation can be found, in case someone wants to understand what this code is doing.

I've also added another commit to remove the line from `cleanup-vendor-files.sh`.

It's ready for another round.

puntope reviewed (approved):
✅  LGTM 

I tested locally
